diff --git a/src/AppBundle/Entity/Battle.php b/src/AppBundle/Entity/Battle.php
index 0c7bef7..5d032d1 100644
--- a/src/AppBundle/Entity/Battle.php
+++ b/src/AppBundle/Entity/Battle.php
@@ -110,4 +110,22 @@ class Battle
     {
         return $this->notes;
     }
+
+    /**
+     * @Serializer\VirtualProperty()
+     * @Serializer\SerializedName("programmer")
+     */
+    public function getProgrammerNickname()
+    {
+        return $this->programmer->getNickname();
+    }
+
+    /**
+     * @Serializer\VirtualProperty()
+     * @Serializer\SerializedName("project")
+     */
+    public function getProjectId()
+    {
+        return $this->project->getId();
+    }
 }
diff --git a/tests/AppBundle/Controller/Api/BattleControllerTest.php b/tests/AppBundle/Controller/Api/BattleControllerTest.php
index 7cf444e..b3707fc 100644
--- a/tests/AppBundle/Controller/Api/BattleControllerTest.php
+++ b/tests/AppBundle/Controller/Api/BattleControllerTest.php
@@ -33,7 +33,11 @@ class BattleControllerTest extends ApiTestCase
         $this->assertEquals(201, $response->getStatusCode());
         $this->asserter()
             ->assertResponsePropertyExists($response, 'didProgrammerWin');
-        $this->debugResponse($response);
+        $this->asserter()
+            ->assertResponsePropertyEquals($response, 'project', $project->getId());
+        $this->asserter()
+            ->assertResponsePropertyEquals($response, 'programmer', 'fred');
+
         // todo for later
         //$this->assertTrue($response->hasHeader('Location'));
     }
