diff --git a/src/AppBundle/Entity/Programmer.php b/src/AppBundle/Entity/Programmer.php
index ee358f4..0cdb294 100644
--- a/src/AppBundle/Entity/Programmer.php
+++ b/src/AppBundle/Entity/Programmer.php
@@ -31,6 +31,8 @@ class Programmer
     private $id;
 
     /**
+     * The unique name of the programmer.
+     *
      * @var string
      *
      * @ORM\Column(name="nickname", type="string", length=100, unique=true)
