diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 23615a1..db11211 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -29,7 +29,10 @@ class ProgrammerController extends BaseController
      * @ApiDoc(
      *  resource=true,
      *  description="Add a new programmer",
-     *  input="AppBundle\Form\ProgrammerType",
+     *  input={
+     *      class="AppBundle\Form\ProgrammerType",
+     *      name=""
+     *  },
      *  output="AppBundle\Entity\Programmer"
      * )
      */
