diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 7363864..080fdc3 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -94,12 +94,28 @@ class ProgrammerController extends BaseController
     }
 
     /**
+     * Returns an array of `Programmer` resources:
+     *
+     * ```
+     * {
+     *     "programmers": []
+     * }
+     * ```
+     *
+     * And here's a Guzzle example:
+     *
+     * ```
+     * $request = $client->get('/api/programmers');
+     * $request = $response->send();
+     * ```
+     *
      * @Route("/api/programmers", name="api_programmers_collection")
      * @Method("GET")
      *
      * @ApiDoc(
      *  description="Return a collection of programmer",
-     *  output="array<AppBundle\Entity\Programmer>"
+     *  output="array<AppBundle\Entity\Programmer>",
+     *  description="Returns a collection of programmers"
      * )
      */
     public function listAction(Request $request)
