diff --git a/src/AppBundle/Form/ProgrammerType.php b/src/AppBundle/Form/ProgrammerType.php
index bc9379b8..cc66e420 100644
--- a/src/AppBundle/Form/ProgrammerType.php
+++ b/src/AppBundle/Form/ProgrammerType.php
@@ -42,6 +42,7 @@ class ProgrammerType extends AbstractType
             'data_class' => 'AppBundle\Entity\Programmer',
             'is_edit' => false,
             'csrf_protection' => false,
+            'extra_fields_message' => 'Extra fields sent! {{ extra_fields }}'
         ));
     }
 }
