diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 61786f9..f32db04 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -66,6 +66,12 @@ class ProgrammerController extends BaseController
     /**
      * @Route("/api/programmers/{nickname}", name="api_programmers_show")
      * @Method("GET")
+     *
+     * @ApiDoc(
+     *  resource=true,
+     *  description="Return a programmer",
+     *  output="AppBundle\Entity\Programmer"
+     * )
      */
     public function showAction($nickname)
     {
@@ -88,6 +94,11 @@ class ProgrammerController extends BaseController
     /**
      * @Route("/api/programmers", name="api_programmers_collection")
      * @Method("GET")
+     *
+     * @ApiDoc(
+     *  description="Return a collection of programmer",
+     *  output="array<AppBundle\Entity\Programmer>"
+     * )
      */
     public function listAction(Request $request)
     {
