diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 1b8ffbd..86c2a54 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -68,6 +68,8 @@ class ProgrammerController extends BaseController
      *  description="Return a programmer",
      *  output="AppBundle\Entity\Programmer"
      * )
+     *
+     * @param string $nickname The unique nickname of your programmer
      */
     public function showAction($nickname)
     {
