diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 5799c6d..a5ed9be 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -73,6 +73,8 @@ class ProgrammerController extends BaseController
      *  description="Return a programmer",
      *  output="AppBundle\Entity\Programmer"
      * )
+     *
+     * @param string $nickname The unique nickname of your programmer
      */
     public function showAction($nickname)
     {
