diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 29c5d64..e10b9ca 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -62,6 +62,12 @@ class ProgrammerController extends BaseController
     /**
      * @Route("/api/programmers/{nickname}", name="api_programmers_show")
      * @Method("GET")
+     *
+     * @ApiDoc(
+     *  resource=true,
+     *  description="Return a programmer",
+     *  output="AppBundle\Entity\Programmer"
+     * )
      */
     public function showAction($nickname)
     {
@@ -84,6 +90,11 @@ class ProgrammerController extends BaseController
     /**
      * @Route("/api/programmers")
      * @Method("GET")
+     *
+     * @ApiDoc(
+     *  description="Return a collection of programmer",
+     *  output="array<AppBundle\Entity\Programmer>"
+     * )
      */
     public function listAction()
     {
