diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 415c2f2..1a35769 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -25,6 +25,11 @@ class ProgrammerController extends BaseController
         $form = $this->createForm(new ProgrammerType(), $programmer);
         $this->processForm($request, $form);
 
+        if (!$form->isValid()) {
+            header('Content-Type: cli');
+            dump((string) $form->getErrors(true, false));die;
+        }
+
         $programmer->setUser($this->findUserByUsername('weaverryan'));
 
         $em = $this->getDoctrine()->getManager();
diff --git a/src/AppBundle/Entity/Programmer.php b/src/AppBundle/Entity/Programmer.php
index 84c131a..b601dfa 100644
--- a/src/AppBundle/Entity/Programmer.php
+++ b/src/AppBundle/Entity/Programmer.php
@@ -4,6 +4,7 @@ namespace AppBundle\Entity;
 
 use Doctrine\ORM\Mapping as ORM;
 use JMS\Serializer\Annotation as Serializer;
+use Symfony\Component\Validator\Constraints as Assert;
 
 /**
  * Programmer
@@ -28,6 +29,7 @@ class Programmer
      *
      * @ORM\Column(name="nickname", type="string", length=100, unique=true)
      * @Serializer\Expose
+     * @Assert\NotBlank(message="Please enter a clever nickname")
      */
     private $nickname;
 
