diff --git a/src/AppBundle/EventListener/ApiExceptionSubscriber.php b/src/AppBundle/EventListener/ApiExceptionSubscriber.php
index facfea8d..5466d2f6 100644
--- a/src/AppBundle/EventListener/ApiExceptionSubscriber.php
+++ b/src/AppBundle/EventListener/ApiExceptionSubscriber.php
@@ -2,6 +2,12 @@
 
 namespace AppBundle\EventListener;
 
-class ApiExceptionSubscriber
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ApiExceptionSubscriber implements EventSubscriberInterface
 {
+    public static function getSubscribedEvents()
+    {
+        // TODO: Implement getSubscribedEvents() method.
+    }
 }
