diff --git a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
index 1cc2981c..c673504d 100644
--- a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
+++ b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
@@ -149,6 +149,7 @@ class ProgrammerControllerTest extends ApiTestCase
     {
         $invalidBody = <<<EOF
 {
+    "nickname": "JohnnyRobot",
     "avatarNumber" : "2
     "tagLine": "I'm from a test!"
 }
