diff --git a/src/AppBundle/Api/ApiProblemException.php b/src/AppBundle/Api/ApiProblemException.php
index 991d43a..5512dd5 100644
--- a/src/AppBundle/Api/ApiProblemException.php
+++ b/src/AppBundle/Api/ApiProblemException.php
@@ -16,4 +16,9 @@ class ApiProblemException extends HttpException
 
         parent::__construct($statusCode, $message, $previous, $headers, $code);
     }
+
+    public function getApiProblem()
+    {
+        return $this->apiProblem;
+    }
 }
diff --git a/src/AppBundle/EventListener/ApiExceptionListener.php b/src/AppBundle/EventListener/ApiExceptionListener.php
index d68c8e3..5dc9e2c 100644
--- a/src/AppBundle/EventListener/ApiExceptionListener.php
+++ b/src/AppBundle/EventListener/ApiExceptionListener.php
@@ -6,6 +6,7 @@ use AppBundle\Api\ApiProblemException;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
 use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\HttpFoundation\JsonResponse;
 
 class ApiExceptionSubscriber implements EventSubscriberInterface
 {
@@ -15,6 +16,16 @@ class ApiExceptionSubscriber implements EventSubscriberInterface
         if (!$e instanceof ApiProblemException) {
             return;
         }
+
+        $apiProblem = $e->getApiProblem();
+
+        $response = new JsonResponse(
+            $apiProblem->toArray(),
+            $apiProblem->getStatusCode()
+        );
+        $response->headers->set('Content-Type', 'application/problem+json');
+
+        $event->setResponse($response);
     }
 
     public static function getSubscribedEvents()
