diff --git a/src/AppBundle/Api/ApiProblem.php b/src/AppBundle/Api/ApiProblem.php
index 2aa72a78..fba2739a 100644
--- a/src/AppBundle/Api/ApiProblem.php
+++ b/src/AppBundle/Api/ApiProblem.php
@@ -8,9 +8,11 @@ namespace AppBundle\Api;
 class ApiProblem
 {
     const TYPE_VALIDATION_ERROR = 'validation_error';
+    const TYPE_INVALID_REQUEST_BODY_FORMAT = 'invalid_body_format';
 
     private static $titles = array(
-        self::TYPE_VALIDATION_ERROR => 'There was a validation error'
+        self::TYPE_VALIDATION_ERROR => 'There was a validation error',
+        self::TYPE_INVALID_REQUEST_BODY_FORMAT => 'Invalid JSON format sent',
     );
 
     private $statusCode;
diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 63b0210d..ade8a62b 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -143,7 +143,9 @@ class ProgrammerController extends BaseController
     {
         $data = json_decode($request->getContent(), true);
         if ($data === null) {
-            throw new HttpException(400, 'Invalid JSON body!');
+            $apiProblem = new ApiProblem(400, ApiProblem::TYPE_INVALID_REQUEST_BODY_FORMAT);
+
+            // now what?
         }
 
         $clearMissing = $request->getMethod() != 'PATCH';
diff --git a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
index c673504d..d5112fd3 100644
--- a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
+++ b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
@@ -161,6 +161,7 @@ EOF;
         $this->debugResponse($response);
 
         $this->assertEquals(400, $response->getStatusCode());
+        $this->asserter()->assertResponsePropertyEquals($response, 'type', 'invalid_body_format');
     }
 
 }
