diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 0bb0c34..66bc351 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -20,7 +20,13 @@ class ProgrammerController extends BaseController
     public function newAction(Request $request)
     {
         $programmer = new Programmer();
-        $this->handleRequest($request, $programmer);
+        $form = $this->handleRequest($request, $programmer);
+
+        if (!$form->isValid()) {
+            header('Content-Type: cli');
+            dump((string) $form->getErrors(true, false));die;
+            die;
+        }
 
         $em = $this->getDoctrine()->getManager();
         $em->persist($programmer);
@@ -143,6 +149,8 @@ class ProgrammerController extends BaseController
         /** @var Programmer $programmer */
         $programmer = $form->getData();
         $programmer->setUser($this->findUserByUsername('weaverryan'));
+
+        return $form;
     }
 
     private function serializeProgrammer(Programmer $programmer)
diff --git a/src/AppBundle/Entity/Programmer.php b/src/AppBundle/Entity/Programmer.php
index 3172a24..bd24303 100644
--- a/src/AppBundle/Entity/Programmer.php
+++ b/src/AppBundle/Entity/Programmer.php
@@ -3,6 +3,7 @@
 namespace AppBundle\Entity;
 
 use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
 
 /**
  * Programmer
@@ -25,6 +26,7 @@ class Programmer
      * @var string
      *
      * @ORM\Column(name="nickname", type="string", length=100, unique=true)
+     * @Assert\NotBlank(message="Please enter a clever nickname")
      */
     private $nickname;
 
