diff --git a/src/AppBundle/Api/ApiProblemException.php b/src/AppBundle/Api/ApiProblemException.php
index f12de49..b8f4d69 100644
--- a/src/AppBundle/Api/ApiProblemException.php
+++ b/src/AppBundle/Api/ApiProblemException.php
@@ -6,4 +6,12 @@ use Symfony\Component\HttpKernel\Exception\HttpException;
 
 class ApiProblemException extends HttpException
 {
+    private $apiProblem;
+
+    public function __construct(ApiProblem $apiProblem, $statusCode, $message = null, \Exception $previous = null, array $headers = array(), $code = 0)
+    {
+        $this->apiProblem = $apiProblem;
+
+        parent::__construct($statusCode, $message, $previous, $headers, $code);
+    }
 }
