diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 4ca2ddd..71b782b 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -193,6 +193,9 @@ class ProgrammerController extends BaseController
             'errors' => $errors
         ];
 
-        return new JsonResponse($data, 400);
+        $response = new JsonResponse($data, 400);
+        $response->headers->set('Content-Type', 'application/problem+json');
+
+        return $response;
     }
 }
diff --git a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
index 0c22649..ae01e99 100644
--- a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
+++ b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
@@ -141,5 +141,6 @@ class ProgrammerControllerTest extends ApiTestCase
         $this->assertResponsePropertyExists($response, 'errors.nickname');
         $this->assertResponsePropertyEquals($response, 'errors.nickname[0]', 'Please enter a clever nickname');
         $this->assertResponsePropertyDoesNotExist($response, 'errors.avatarNumber');
+        $this->assertEquals('application/problem+json', $response->getHeader('Content-Type'));
     }
 }
