diff --git a/src/AppBundle/EventListener/ApiExceptionListener.php b/src/AppBundle/EventListener/ApiExceptionListener.php
new file mode 100644
index 0000000..facfea8
--- /dev/null
+++ b/src/AppBundle/EventListener/ApiExceptionListener.php
@@ -0,0 +1,7 @@
+<?php
+
+namespace AppBundle\EventListener;
+
+class ApiExceptionSubscriber
+{
+}
diff --git a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
index d5112fd..8ae72e5 100644
--- a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
+++ b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
@@ -158,7 +158,6 @@ EOF;
         $response = $this->client->post('/api/programmers', [
             'body' => $invalidBody
         ]);
-        $this->debugResponse($response);
 
         $this->assertEquals(400, $response->getStatusCode());
         $this->asserter()->assertResponsePropertyEquals($response, 'type', 'invalid_body_format');
