diff --git a/src/AppBundle/Api/ApiProblem.php b/src/AppBundle/Api/ApiProblem.php
index d4e24a2d..2aa72a78 100644
--- a/src/AppBundle/Api/ApiProblem.php
+++ b/src/AppBundle/Api/ApiProblem.php
@@ -9,6 +9,10 @@ class ApiProblem
 {
     const TYPE_VALIDATION_ERROR = 'validation_error';
 
+    private static $titles = array(
+        self::TYPE_VALIDATION_ERROR => 'There was a validation error'
+    );
+
     private $statusCode;
 
     private $type;
@@ -17,11 +21,16 @@ class ApiProblem
 
     private $extraData = array();
 
-    public function __construct($statusCode, $type, $title)
+    public function __construct($statusCode, $type)
     {
         $this->statusCode = $statusCode;
         $this->type = $type;
-        $this->title = $title;
+
+        if (!isset(self::$titles[$type])) {
+            throw new \InvalidArgumentException('No title for type '.$type);
+        }
+
+        $this->title = self::$titles[$type];
     }
 
     public function toArray()
diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 4f78a9e2..95ff4d39 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -170,8 +170,7 @@ class ProgrammerController extends BaseController
 
         $apiProblem = new ApiProblem(
             400,
-            ApiProblem::TYPE_VALIDATION_ERROR,
-            'There was a validation error'
+            ApiProblem::TYPE_VALIDATION_ERROR
         );
         $apiProblem->set('errors', $errors);
 
