diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index e189c7ef..1bf69186 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -26,15 +26,7 @@ class ProgrammerController extends BaseController
         $this->processForm($request, $form);
 
         if (!$form->isValid()) {
-            $errors = $this->getErrorsFromForm($form);
-
-            $data = [
-                'type' => 'validation_error',
-                'title' => 'There was a validation error',
-                'errors' => $errors
-            ];
-
-            return new JsonResponse($data, 400);
+            return $this->createValidationErrorResponse($form);
         }
 
         $programmer->setUser($this->findUserByUsername('weaverryan'));
@@ -110,6 +102,10 @@ class ProgrammerController extends BaseController
         $form = $this->createForm(new UpdateProgrammerType(), $programmer);
         $this->processForm($request, $form);
 
+        if (!$form->isValid()) {
+            return $this->createValidationErrorResponse($form);
+        }
+
         $em = $this->getDoctrine()->getManager();
         $em->persist($programmer);
         $em->flush();
@@ -166,4 +162,17 @@ class ProgrammerController extends BaseController
 
         return $errors;
     }
+
+    private function createValidationErrorResponse(FormInterface $form)
+    {
+        $errors = $this->getErrorsFromForm($form);
+
+        $data = [
+            'type' => 'validation_error',
+            'title' => 'There was a validation error',
+            'errors' => $errors
+        ];
+
+        return new JsonResponse($data, 400);
+    }
 }
