diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index af76b06..00301ac 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -57,16 +57,41 @@ class ProgrammerController extends BaseController
             ));
         }
 
-        $data = array(
-            'nickname' => $programmer->getNickname(),
-            'avatarNumber' => $programmer->getAvatarNumber(),
-            'powerLevel' => $programmer->getPowerLevel(),
-            'tagLine' => $programmer->getTagLine(),
-        );
+        $data = $this->serializeProgrammer($programmer);
 
         $response = new Response(json_encode($data), 200);
         $response->headers->set('Content-Type', 'application/json');
 
         return $response;
     }
+
+    /**
+     * @Route("/api/programmers")
+     * @Method("GET")
+     */
+    public function listAction()
+    {
+        $programmers = $this->getDoctrine()
+            ->getRepository('AppBundle:Programmer')
+            ->findAll();
+        $data = array('programmers' => array());
+        foreach ($programmers as $programmer) {
+            $data['programmers'][] = $this->serializeProgrammer($programmer);
+        }
+
+        $response = new Response(json_encode($data), 200);
+        $response->headers->set('Content-Type', 'application/json');
+
+        return $response;
+    }
+
+    private function serializeProgrammer(Programmer $programmer)
+    {
+        return array(
+            'nickname' => $programmer->getNickname(),
+            'avatarNumber' => $programmer->getAvatarNumber(),
+            'powerLevel' => $programmer->getPowerLevel(),
+            'tagLine' => $programmer->getTagLine(),
+        );
+    }
 }
diff --git a/testing.php b/testing.php
index 05c13f9..11139ad 100644
--- a/testing.php
+++ b/testing.php
@@ -26,5 +26,7 @@ $programmerUrl = $response->getHeader('Location');
 // 2) GET a programmer resource
 $response = $client->get($programmerUrl);
 
+$response = $client->get('/api/programmers');
+
 echo $response;
 echo "\n\n";
