diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 84f2f86..cb15c6d 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -7,6 +7,7 @@ use AppBundle\Entity\Programmer;
 use AppBundle\Form\ProgrammerType;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 
@@ -31,13 +32,12 @@ class ProgrammerController extends BaseController
         $em->flush();
 
         $data = $this->serializeProgrammer($programmer);
-        $response = new Response(json_encode($data), 201);
+        $response = new JsonResponse($data, 201);
         $programmerUrl = $this->generateUrl(
             'api_programmers_show',
             ['nickname' => $programmer->getNickname()]
         );
         $response->headers->set('Location', $programmerUrl);
-        $response->headers->set('Content-Type', 'application/json');
 
         return $response;
     }
@@ -60,8 +60,7 @@ class ProgrammerController extends BaseController
 
         $data = $this->serializeProgrammer($programmer);
 
-        $response = new Response(json_encode($data), 200);
-        $response->headers->set('Content-Type', 'application/json');
+        $response = new JsonResponse($data, 200);
 
         return $response;
     }
@@ -79,8 +78,7 @@ class ProgrammerController extends BaseController
             $data['programmers'][] = $this->serializeProgrammer($programmer);
         }
 
-        $response = new Response(json_encode($data), 200);
-        $response->headers->set('Content-Type', 'application/json');
+        $response = new JsonResponse($data, 200);
 
         return $response;
     }
