diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index c271b90..da688a2 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -45,6 +45,13 @@ class ProgrammerController extends BaseController
             ->getRepository('AppBundle:Programmer')
             ->findOneByNickname($nickname);
 
+        if (!$programmer) {
+            throw $this->createNotFoundException(sprintf(
+                'No programmer found with nickname "%s"',
+                $nickname
+            ));
+        }
+
         $data = array(
             'nickname' => $programmer->getNickname(),
             'avatarNumber' => $programmer->getAvatarNumber(),
diff --git a/testing.php b/testing.php
index b9dbd1d..8fb6898 100644
--- a/testing.php
+++ b/testing.php
@@ -22,7 +22,7 @@ $response = $client->post('/api/programmers', [
 ]);
 
 // 2) GET a programmer resource
-$response = $client->get('/api/programmers/'.$nickname);
+$response = $client->get('/api/programmers/abcd'.$nickname);
 
 echo $response;
 echo "\n\n";
