diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 8c424ef..af76b06 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -31,13 +31,17 @@ class ProgrammerController extends BaseController
         $em->flush();
 
         $response = new Response('It worked. Believe me - I\'m an API', 201);
-        $response->headers->set('Location', '/some/programmer/url');
+        $programmerUrl = $this->generateUrl(
+            'api_programmers_show',
+            ['nickname' => $programmer->getNickname()]
+        );
+        $response->headers->set('Location', $programmerUrl);
 
         return $response;
     }
 
     /**
-     * @Route("/api/programmers/{nickname}")
+     * @Route("/api/programmers/{nickname}", name="api_programmers_show")
      * @Method("GET")
      */
     public function showAction($nickname)
diff --git a/testing.php b/testing.php
index 8fb6898..ed4f9cb 100644
--- a/testing.php
+++ b/testing.php
@@ -21,6 +21,10 @@ $response = $client->post('/api/programmers', [
     'body' => json_encode($data)
 ]);
 
+echo $response;
+echo "\n\n";
+die;
+
 // 2) GET a programmer resource
 $response = $client->get('/api/programmers/abcd'.$nickname);
 
