diff --git a/src/AppBundle/Entity/Battle.php b/src/AppBundle/Entity/Battle.php
index c9ebe054..21d5370d 100644
--- a/src/AppBundle/Entity/Battle.php
+++ b/src/AppBundle/Entity/Battle.php
@@ -17,7 +17,8 @@ use Hateoas\Configuration\Annotation as Hateoas;
  *     href=@Hateoas\Route(
  *          "api_programmers_show",
  *          parameters={"nickname"= "expr(object.getProgrammerNickname())"}
- *     )
+ *     ),
+ *     embedded = "expr(object.getProgrammer())"
  * )
  */
 class Battle
diff --git a/tests/AppBundle/Controller/Api/BattleControllerTest.php b/tests/AppBundle/Controller/Api/BattleControllerTest.php
index 22403c45..becfb20d 100644
--- a/tests/AppBundle/Controller/Api/BattleControllerTest.php
+++ b/tests/AppBundle/Controller/Api/BattleControllerTest.php
@@ -43,6 +43,7 @@ class BattleControllerTest extends ApiTestCase
             '_links.programmer.href',
             $this->adjustUri('/api/programmers/Fred')
         );
+        $this->debugResponse($response);
 
         // todo for later
         //$this->assertTrue($response->hasHeader('Location'));
