diff --git a/src/AppBundle/Entity/Programmer.php b/src/AppBundle/Entity/Programmer.php
index ee358f4..2c67325 100644
--- a/src/AppBundle/Entity/Programmer.php
+++ b/src/AppBundle/Entity/Programmer.php
@@ -6,6 +6,7 @@ use Doctrine\ORM\Mapping as ORM;
 use JMS\Serializer\Annotation as Serializer;
 use Symfony\Component\Validator\Constraints as Assert;
 use AppBundle\Annotation\Link;
+use Hateoas\Configuration\Annotation as Hateoas;
 
 /**
  * Programmer
@@ -13,10 +14,12 @@ use AppBundle\Annotation\Link;
  * @ORM\Table(name="battle_programmer")
  * @ORM\Entity(repositoryClass="AppBundle\Repository\ProgrammerRepository")
  * @Serializer\ExclusionPolicy("all")
- * @Link(
- *  "self",
- *  route = "api_programmers_show",
- *  params = { "nickname": "object.getNickname()" }
+ * @Hateoas\Relation(
+ *      "self",
+ *      href=@Hateoas\Route(
+ *          "api_programmers_show",
+ *          parameters = { "nickname"= "expr(object.getNickname())" }
+ *      )
  * )
  */
 class Programmer
diff --git a/tests/AppBundle/Controller/Api/ProgrammerControllerTest.php b/tests/AppBundle/Controller/Api/ProgrammerControllerTest.php
index 107fb34..80c2ba6 100644
--- a/tests/AppBundle/Controller/Api/ProgrammerControllerTest.php
+++ b/tests/AppBundle/Controller/Api/ProgrammerControllerTest.php
@@ -55,7 +55,7 @@ class ProgrammerControllerTest extends ApiTestCase
         $this->asserter()->assertResponsePropertyEquals($response, 'nickname', 'UnitTester');
         $this->asserter()->assertResponsePropertyEquals(
             $response,
-            '_links.self',
+            '_links.self.href',
             $this->adjustUri('/api/programmers/UnitTester')
         );
     }
