diff --git a/src/AppBundle/Entity/Battle.php b/src/AppBundle/Entity/Battle.php
index 93ff962..c9ebe05 100644
--- a/src/AppBundle/Entity/Battle.php
+++ b/src/AppBundle/Entity/Battle.php
@@ -6,15 +6,18 @@ use AppBundle\Annotation\Link;
 use Doctrine\ORM\Mapping as ORM;
 
 use JMS\Serializer\Annotation as Serializer;
+use Hateoas\Configuration\Annotation as Hateoas;
 
 /**
  * @ORM\Table(name="battle_battle")
  * @ORM\Entity(repositoryClass="AppBundle\Repository\BattleRepository")
  * @Serializer\ExclusionPolicy("all")
- * @Link(
+ * @Hateoas\Relation(
  *     "programmer",
- *     route="api_programmers_show",
- *     params={"nickname": "object.getProgrammerNickname()"}
+ *     href=@Hateoas\Route(
+ *          "api_programmers_show",
+ *          parameters={"nickname"= "expr(object.getProgrammerNickname())"}
+ *     )
  * )
  */
 class Battle
diff --git a/tests/AppBundle/Controller/Api/BattleControllerTest.php b/tests/AppBundle/Controller/Api/BattleControllerTest.php
index 2951b4d..6adbf7c 100644
--- a/tests/AppBundle/Controller/Api/BattleControllerTest.php
+++ b/tests/AppBundle/Controller/Api/BattleControllerTest.php
@@ -40,7 +40,7 @@ class BattleControllerTest extends ApiTestCase
 
         $this->asserter()->assertResponsePropertyEquals(
             $response,
-            '_links.programmer',
+            '_links.programmer.href',
             $this->adjustUri('/api/programmers/fred')
         );
 
