diff --git a/src/AppBundle/Security/JwtTokenAuthenticator.php b/src/AppBundle/Security/JwtTokenAuthenticator.php
new file mode 100644
index 0000000..6d02142
--- /dev/null
+++ b/src/AppBundle/Security/JwtTokenAuthenticator.php
@@ -0,0 +1,62 @@
+<?php
+
+namespace AppBundle\Security;
+
+use Lexik\Bundle\JWTAuthenticationBundle\TokenExtractor\AuthorizationHeaderTokenExtractor;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Guard\AbstractGuardAuthenticator;
+
+class JwtTokenAuthenticator extends AbstractGuardAuthenticator
+{
+    public function getCredentials(Request $request)
+    {
+        $extractor = new AuthorizationHeaderTokenExtractor(
+            'Bearer',
+            'Authorization'
+        );
+
+        $token = $extractor->extract($request);
+
+        if (!$token) {
+            return;
+        }
+
+        return $token;
+
+    }
+
+    public function getUser($credentials, UserProviderInterface $userProvider)
+    {
+        // TODO: Implement getUser() method.
+    }
+
+    public function checkCredentials($credentials, UserInterface $user)
+    {
+        // TODO: Implement checkCredentials() method.
+    }
+
+    public function onAuthenticationFailure(Request $request, AuthenticationException $exception)
+    {
+        // TODO: Implement onAuthenticationFailure() method.
+    }
+
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token, $providerKey)
+    {
+        // TODO: Implement onAuthenticationSuccess() method.
+    }
+
+    public function supportsRememberMe()
+    {
+        // TODO: Implement supportsRememberMe() method.
+    }
+
+    public function start(Request $request, AuthenticationException $authException = null)
+    {
+        // TODO: Implement start() method.
+    }
+}
