diff --git a/tests/AppBundle/Controller/Api/TokenControllerTest.php b/tests/AppBundle/Controller/Api/TokenControllerTest.php
index 5115564..0101b1b 100644
--- a/tests/AppBundle/Controller/Api/TokenControllerTest.php
+++ b/tests/AppBundle/Controller/Api/TokenControllerTest.php
@@ -19,4 +19,14 @@ class TokenControllerTest extends ApiTestCase
             'token'
         );
     }
+
+    public function testPOSTTokenInvalidCredentials()
+    {
+        $this->createUser('weaverryan', 'I<3Pizza');
+
+        $response = $this->client->post('/api/tokens', [
+            'auth' => ['weaverryan', 'IH8Pizza']
+        ]);
+        $this->assertEquals(401, $response->getStatusCode());
+    }
 }
