diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 4f7e7d6..04c9ef6 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -8,6 +8,8 @@ use AppBundle\Controller\BaseController;
 use AppBundle\Entity\Programmer;
 use AppBundle\Form\ProgrammerType;
 use AppBundle\Form\UpdateProgrammerType;
+use Pagerfanta\Adapter\DoctrineORMAdapter;
+use Pagerfanta\Pagerfanta;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
 use Symfony\Component\Form\FormInterface;
@@ -74,13 +76,28 @@ class ProgrammerController extends BaseController
      * @Route("/api/programmers")
      * @Method("GET")
      */
-    public function listAction()
+    public function listAction(Request $request)
     {
-        $programmers = $this->getDoctrine()
+        $page = $request->query->get('page', 1);
+
+        $qb = $this->getDoctrine()
             ->getRepository('AppBundle:Programmer')
-            ->findAll();
+            ->findAllQueryBuilder();
+        $adapter = new DoctrineORMAdapter($qb);
+        $pagerfanta = new Pagerfanta($adapter);
+        $pagerfanta->setMaxPerPage(10);
+        $pagerfanta->setCurrentPage($page);
+
+        $programmers = [];
+        foreach ($pagerfanta->getCurrentPageResults() as $result) {
+            $programmers[] = $result;
+        }
 
-        $response = $this->createApiResponse(['programmers' => $programmers], 200);
+        $response = $this->createApiResponse([
+            'total' => $pagerfanta->getNbResults(),
+            'count' => count($programmers),
+            'programmers' => $programmers,
+        ], 200);
 
         return $response;
     }
diff --git a/src/AppBundle/Repository/ProgrammerRepository.php b/src/AppBundle/Repository/ProgrammerRepository.php
index ead7472..ed7ff64 100644
--- a/src/AppBundle/Repository/ProgrammerRepository.php
+++ b/src/AppBundle/Repository/ProgrammerRepository.php
@@ -25,4 +25,9 @@ class ProgrammerRepository extends EntityRepository
     {
         return $this->findOneBy(array('nickname' => $nickname));
     }
+
+    public function findAllQueryBuilder()
+    {
+        return $this->createQueryBuilder('programmer');
+    }
 }
