diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 8b65c8a..0bb0c34 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -82,7 +82,7 @@ class ProgrammerController extends BaseController
 
     /**
      * @Route("/api/programmers/{nickname}")
-     * @Method("PUT")
+     * @Method({"PUT", "PATCH"})
      */
     public function updateAction($nickname, Request $request)
     {
@@ -135,9 +135,10 @@ class ProgrammerController extends BaseController
     {
         $data = json_decode($request->getContent(), true);
         $form = $this->createForm(new ProgrammerType(), $programmer, [
-            'is_edit' => $programmer && $programmer->getId()
+            'is_edit' => $programmer->getId()
         ]);
-        $form->submit($data);
+        $clearMissing = $request->getMethod() != 'PATCH';
+        $form->submit($data, $clearMissing);
 
         /** @var Programmer $programmer */
         $programmer = $form->getData();
