diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 0340b87f..cc3702be 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -87,7 +87,7 @@ class ProgrammerController extends BaseController
 
     /**
      * @Route("/api/programmers/{nickname}")
-     * @Method("PUT")
+     * @Method({"PUT", "PATCH"})
      */
     public function updateAction($nickname, Request $request)
     {
@@ -140,7 +140,9 @@ class ProgrammerController extends BaseController
     private function processForm(Request $request, FormInterface $form)
     {
         $data = json_decode($request->getContent(), true);
-        $form->submit($data);
+
+        $clearMissing = $request->getMethod() != 'PATCH';
+        $form->submit($data, $clearMissing);
     }
 
     private function serializeProgrammer(Programmer $programmer)
