diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 7eb131b..98292d8 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -30,6 +30,6 @@ class ProgrammerController extends BaseController
         $em->persist($programmer);
         $em->flush();
 
-        return new Response('It worked. Believe me - I\'m an API');
+        return new Response('It worked. Believe me - I\'m an API', 201);
     }
 }
