diff --git a/testing.php b/testing.php
index 2e05cdd2..f935694c 100644
--- a/testing.php
+++ b/testing.php
@@ -9,7 +9,16 @@ $client = new \GuzzleHttp\Client([
     ]
 ]);
 
-$response = $client->post('/api/programmers');
+$nickname = 'ObjectOrienter'.rand(0, 999);
+$data = array(
+    'nickname' => $nickname,
+    'avatarNumber' => 5,
+    'tagLine' => 'a test dev!'
+);
+
+$response = $client->post('/api/programmers', [
+    'body' => json_encode($data)
+]);
 
 echo $response;
 echo "\n\n";
