diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index 862ac38..3891d71 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -19,14 +19,8 @@ class ProgrammerController extends BaseController
      */
     public function newAction(Request $request)
     {
-        $data = json_decode($request->getContent(), true);
-
-        $form = $this->createForm(new ProgrammerType());
-        $form->submit($data);
-
-        /** @var Programmer $programmer */
-        $programmer = $form->getData();
-        $programmer->setUser($this->findUserByUsername('weaverryan'));
+        $programmer = new Programmer();
+        $this->handleRequest($request, $programmer);
 
         $em = $this->getDoctrine()->getManager();
         $em->persist($programmer);
@@ -103,13 +97,7 @@ class ProgrammerController extends BaseController
             ));
         }
 
-        $data = json_decode($request->getContent(), true);
-        $form = $this->createForm(new ProgrammerType(), $programmer);
-        $form->submit($data);
-
-        /** @var Programmer $programmer */
-        $programmer = $form->getData();
-        $programmer->setUser($this->findUserByUsername('weaverryan'));
+        $this->handleRequest($request, $programmer);
 
         $em = $this->getDoctrine()->getManager();
         $em->persist($programmer);
@@ -121,6 +109,16 @@ class ProgrammerController extends BaseController
         return $response;
     }
 
+    private function handleRequest(Request $request, Programmer $programmer)
+    {
+        $data = json_decode($request->getContent(), true);
+        $form = $this->createForm(new ProgrammerType(), $programmer);
+        $form->submit($data);
+
+        /** @var Programmer $programmer */
+        $programmer = $form->getData();
+        $programmer->setUser($this->findUserByUsername('weaverryan'));
+    }
 
     private function serializeProgrammer(Programmer $programmer)
     {
