diff --git a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
index 1db7e1af..62270c76 100644
--- a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
+++ b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
@@ -68,4 +68,24 @@ class ProgrammerControllerTest extends ApiTestCase
         $this->asserter()->assertResponsePropertyCount($response, 'programmers', 2);
         $this->asserter()->assertResponsePropertyEquals($response, 'programmers[1].nickname', 'CowboyCoder');
     }
+
+    public function testPUTProgrammer()
+    {
+        $this->createProgrammer(array(
+            'nickname' => 'CowboyCoder',
+            'avatarNumber' => 5,
+            'tagLine' => 'foo',
+        ));
+
+        $data = array(
+            'nickname' => 'CowboyCoder',
+            'avatarNumber' => 2,
+            'tagLine' => 'foo',
+        );
+        $response = $this->client->put('/api/programmers/CowboyCoder', [
+            'body' => json_encode($data)
+        ]);
+        $this->assertEquals(200, $response->getStatusCode());
+        $this->asserter()->assertResponsePropertyEquals($response, 'avatarNumber', 2);
+    }
 }
