diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index cb15c6d..4064571 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -44,6 +44,7 @@ class ProgrammerController extends BaseController
 
     /**
      * @Route("/api/programmers/{nickname}", name="api_programmers_show")
+     * @Method("GET")
      */
     public function showAction($nickname)
     {
@@ -67,6 +68,7 @@ class ProgrammerController extends BaseController
 
     /**
      * @Route("/api/programmers")
+     * @Method("GET")
      */
     public function listAction()
     {
@@ -83,6 +85,42 @@ class ProgrammerController extends BaseController
         return $response;
     }
 
+    /**
+     * @Route("/api/programmers/{nickname}")
+     * @Method("PUT")
+     */
+    public function updateAction($nickname, Request $request)
+    {
+        $programmer = $this->getDoctrine()
+            ->getRepository('AppBundle:Programmer')
+            ->findOneByNickname($nickname);
+
+        if (!$programmer) {
+            throw $this->createNotFoundException(sprintf(
+                'No programmer found with nickname "%s"',
+                $nickname
+            ));
+        }
+
+        $data = json_decode($request->getContent(), true);
+        $form = $this->createForm(new ProgrammerType(), $programmer);
+        $form->submit($data);
+
+        /** @var Programmer $programmer */
+        $programmer = $form->getData();
+        $programmer->setUser($this->findUserByUsername('weaverryan'));
+
+        $em = $this->getDoctrine()->getManager();
+        $em->persist($programmer);
+        $em->flush();
+
+        $data = $this->serializeProgrammer($programmer);
+        $response = new JsonResponse($data, 200);
+
+        return $response;
+    }
+
+
     private function serializeProgrammer(Programmer $programmer)
     {
         return array(
