diff --git a/src/AppBundle/Controller/Api/ProgrammerController.php b/src/AppBundle/Controller/Api/ProgrammerController.php
index cc3702be..23fe4f4d 100644
--- a/src/AppBundle/Controller/Api/ProgrammerController.php
+++ b/src/AppBundle/Controller/Api/ProgrammerController.php
@@ -31,8 +31,8 @@ class ProgrammerController extends BaseController
         $em->persist($programmer);
         $em->flush();
 
-        $data = $this->serializeProgrammer($programmer);
-        $response = new JsonResponse($data, 201);
+        $json = $this->serialize($programmer);
+        $response = new Response($json, 201);
         $programmerUrl = $this->generateUrl(
             'api_programmers_show',
             ['nickname' => $programmer->getNickname()]
@@ -59,9 +59,9 @@ class ProgrammerController extends BaseController
             ));
         }
 
-        $data = $this->serializeProgrammer($programmer);
+        $json = $this->serialize($programmer);
 
-        $response = new JsonResponse($data, 200);
+        $response = new Response($json, 200);
 
         return $response;
     }
@@ -75,12 +75,9 @@ class ProgrammerController extends BaseController
         $programmers = $this->getDoctrine()
             ->getRepository('AppBundle:Programmer')
             ->findAll();
-        $data = array('programmers' => array());
-        foreach ($programmers as $programmer) {
-            $data['programmers'][] = $this->serializeProgrammer($programmer);
-        }
+        $json = $this->serialize(['programmers' => $programmers]);
 
-        $response = new JsonResponse($data, 200);
+        $response = new Response($json, 200);
 
         return $response;
     }
@@ -109,8 +106,8 @@ class ProgrammerController extends BaseController
         $em->persist($programmer);
         $em->flush();
 
-        $data = $this->serializeProgrammer($programmer);
-        $response = new JsonResponse($data, 200);
+        $json = $this->serialize($programmer);
+        $response = new Response($json, 200);
 
         return $response;
     }
@@ -145,13 +142,9 @@ class ProgrammerController extends BaseController
         $form->submit($data, $clearMissing);
     }
 
-    private function serializeProgrammer(Programmer $programmer)
+    private function serialize($data)
     {
-        return array(
-            'nickname' => $programmer->getNickname(),
-            'avatarNumber' => $programmer->getAvatarNumber(),
-            'powerLevel' => $programmer->getPowerLevel(),
-            'tagLine' => $programmer->getTagLine(),
-        );
+        return $this->container->get('jms_serializer')
+            ->serialize($data, 'json');
     }
 }
