diff --git a/src/AppBundle/Entity/Programmer.php b/src/AppBundle/Entity/Programmer.php
index 3172a24..2406613 100644
--- a/src/AppBundle/Entity/Programmer.php
+++ b/src/AppBundle/Entity/Programmer.php
@@ -3,12 +3,14 @@
 namespace AppBundle\Entity;
 
 use Doctrine\ORM\Mapping as ORM;
+use JMS\Serializer\Annotation as Serializer;
 
 /**
  * Programmer
  *
  * @ORM\Table(name="battle_programmer")
  * @ORM\Entity(repositoryClass="AppBundle\Repository\ProgrammerRepository")
+ * @Serializer\ExclusionPolicy("all")
  */
 class Programmer
 {
@@ -18,6 +20,7 @@ class Programmer
      * @ORM\Column(name="id", type="integer")
      * @ORM\Id
      * @ORM\GeneratedValue(strategy="AUTO")
+     * @Serializer\Expose
      */
     private $id;
 
@@ -25,6 +28,7 @@ class Programmer
      * @var string
      *
      * @ORM\Column(name="nickname", type="string", length=100, unique=true)
+     * @Serializer\Expose
      */
     private $nickname;
 
@@ -32,6 +36,7 @@ class Programmer
      * @var integer
      *
      * @ORM\Column(name="avatarNumber", type="integer")
+     * @Serializer\Expose
      */
     private $avatarNumber;
 
@@ -39,6 +44,7 @@ class Programmer
      * @var string
      *
      * @ORM\Column(name="tagLine", type="string", length=255, nullable=true)
+     * @Serializer\Expose
      */
     private $tagLine;
 
@@ -46,6 +52,7 @@ class Programmer
      * @var integer
      *
      * @ORM\Column(name="powerLevel", type="integer")
+     * @Serializer\Expose
      */
     private $powerLevel = 0;
 
diff --git a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
index b1f5b03..59d79dd 100644
--- a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
+++ b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
@@ -47,6 +47,7 @@ class ProgrammerControllerTest extends ApiTestCase
             'tagLine'
         ));
         $this->assertResponsePropertyEquals($response, 'nickname', 'UnitTester');
+        $this->assertResponsePropertyEquals($response, 'nickname', 'UnitTester');
     }
 
     public function testGETProgrammersCollection()
