diff --git a/src/AppBundle/Tests/Controller/Api/ApiTestCase.php b/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
index 9b17bfb..7c198d5 100644
--- a/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
+++ b/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
@@ -2,7 +2,9 @@
 
 namespace AppBundle\Tests\Controller\Api;
 
+use AppBundle\Entity\User;
 use Doctrine\Common\DataFixtures\Purger\ORMPurger;
+use Doctrine\ORM\EntityManager;
 use Exception;
 use GuzzleHttp\Client;
 use GuzzleHttp\Message\AbstractMessage;
@@ -205,4 +207,29 @@ class ApiTestCase extends KernelTestCase
 
         $this->printDebug($output);
     }
+
+    protected function createUser($username, $plainPassword = 'foo')
+    {
+        $user = new User();
+        $user->setUsername($username);
+        $user->setEmail($username.'@foo.com');
+        $password = $this->getService('security.password_encoder')
+            ->encodePassword($user, $plainPassword);
+        $user->setPassword($password);
+
+        $em = $this->getEntityManager();
+        $em->persist($user);
+        $em->flush($user);
+
+        return $user;
+    }
+
+    /**
+     * @return EntityManager
+     */
+    protected function getEntityManager()
+    {
+        return $this->getService('doctrine')
+            ->getManager();
+    }
 }
diff --git a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
index b774b2f..206e688 100644
--- a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
+++ b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
@@ -3,6 +3,13 @@ namespace AppBundle\Tests\Controller\Api;
 
 class ProgrammerControllerTest extends ApiTestCase
 {
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->createUser('weaverryan');
+    }
+
     public function testPOST()
     {
         $data = array(
