diff --git a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
index 9144784..0e1a80c 100644
--- a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
+++ b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
@@ -23,5 +23,10 @@ class ProgrammerControllerTest extends \PHPUnit_Framework_TestCase
         $response = $client->post('/api/programmers', [
             'body' => json_encode($data)
         ]);
+
+        $this->assertEquals(201, $response->getStatusCode());
+        $this->assertTrue($response->hasHeader('Location'));
+        $data = json_decode($response->getBody(true), true);
+        $this->assertArrayHasKey('nickname', $data);
     }
 }
