diff --git a/src/AppBundle/Tests/Controller/Api/ApiTestCase.php b/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
index eb87824..8e21d32 100644
--- a/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
+++ b/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
@@ -8,6 +8,7 @@ use GuzzleHttp\Client;
 use GuzzleHttp\Message\ResponseInterface;
 use GuzzleHttp\Subscriber\History;
 use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\Console\Output\ConsoleOutput;
 use Symfony\Component\DomCrawler\Crawler;
 
 class ApiTestCase extends KernelTestCase
@@ -24,6 +25,8 @@ class ApiTestCase extends KernelTestCase
      */
     protected $client;
 
+    private $output;
+
     public static function setUpBeforeClass()
     {
         self::$staticClient = new Client([
@@ -114,6 +117,10 @@ class ApiTestCase extends KernelTestCase
 
     protected function printDebug($string)
     {
-        echo $string."\n";
+        if ($this->output === null) {
+            $this->output = new ConsoleOutput();
+        }
+
+        $this->output->writeln($string);
     }
 }
\ No newline at end of file
