diff --git a/src/AppBundle/Tests/Controller/Api/ApiTestCase.php b/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
index d866b34..b03f476 100644
--- a/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
+++ b/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
@@ -8,6 +8,7 @@ use Doctrine\Common\DataFixtures\Purger\ORMPurger;
 use Doctrine\ORM\EntityManager;
 use Exception;
 use GuzzleHttp\Client;
+use GuzzleHttp\Event\BeforeEvent;
 use GuzzleHttp\Message\AbstractMessage;
 use GuzzleHttp\Message\ResponseInterface;
 use GuzzleHttp\Subscriber\History;
@@ -62,6 +63,15 @@ class ApiTestCase extends KernelTestCase
         self::$staticClient->getEmitter()
             ->attach(self::$history);
 
+        // guaranteeing that /app_test.php is prefixed to all URLs
+        self::$staticClient->getEmitter()
+            ->on('before', function(BeforeEvent $event) {
+                $path = $event->getRequest()->getPath();
+                if (strpos($path, '/api') === 0) {
+                    $event->getRequest()->setPath('/app_test.php'.$path);
+                }
+            });
+
         self::bootKernel();
     }
 
diff --git a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
index 4c4b0c7..0a6bca7 100644
--- a/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
+++ b/src/AppBundle/Tests/Controller/Api/ProgrammerControllerTest.php
@@ -22,11 +22,10 @@ class ProgrammerControllerTest extends ApiTestCase
         $response = $this->client->post('/api/programmers', [
             'body' => json_encode($data)
         ]);
-        $this->printLastRequestUrl();
 
         $this->assertEquals(201, $response->getStatusCode());
         $this->assertTrue($response->hasHeader('Location'));
-        $this->assertEquals('/api/programmers/ObjectOrienter', $response->getHeader('Location'));
+        $this->assertStringEndsWith('/api/programmers/ObjectOrienter', $response->getHeader('Location'));
         $data = json_decode($response->getBody(true), true);
         $this->assertArrayHasKey('nickname', $data);
         $this->assertEquals('ObjectOrienter', $data['nickname']);
diff --git a/web/app_test.php b/web/app_test.php
index 503a56d..0bed2d7 100644
--- a/web/app_test.php
+++ b/web/app_test.php
@@ -1,5 +1,5 @@
 <?php
-die('working?');
+
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Debug\Debug;
 
