diff --git a/app/phpunit.xml.dist b/app/phpunit.xml.dist
index 2b41503..6090fdd 100644
--- a/app/phpunit.xml.dist
+++ b/app/phpunit.xml.dist
@@ -15,11 +15,9 @@
         </testsuite>
     </testsuites>
 
-    <!--
     <php>
-        <server name="KERNEL_DIR" value="/path/to/your/app/" />
+        <env name="TEST_BASE_URL" value="http://localhost:8000" />
     </php>
-    -->
 
     <filter>
         <whitelist>
diff --git a/src/AppBundle/Test/ApiTestCase.php b/src/AppBundle/Test/ApiTestCase.php
index 7636f91..f87e5ac 100644
--- a/src/AppBundle/Test/ApiTestCase.php
+++ b/src/AppBundle/Test/ApiTestCase.php
@@ -45,8 +45,9 @@ class ApiTestCase extends KernelTestCase
 
     public static function setUpBeforeClass()
     {
+        $baseUrl = getenv('TEST_BASE_URL');
         self::$staticClient = new Client([
-            'base_url' => 'http://localhost:8000',
+            'base_url' => $baseUrl,
             'defaults' => [
                 'exceptions' => false
             ]
