diff --git a/src/AppBundle/Test/ApiTestCase.php b/src/AppBundle/Test/ApiTestCase.php
index cd06e80..1abee37 100644
--- a/src/AppBundle/Test/ApiTestCase.php
+++ b/src/AppBundle/Test/ApiTestCase.php
@@ -27,6 +27,11 @@ class ApiTestCase extends KernelTestCase
      */
     protected $client;
 
+    /**
+     * @var ConsoleOutput
+     */
+    private $output;
+
     /**
      * @var FormatterHelper
      */
@@ -184,7 +189,11 @@ class ApiTestCase extends KernelTestCase
      */
     protected function printDebug($string)
     {
-        echo $string."\n";
+        if ($this->output === null) {
+            $this->output = new ConsoleOutput();
+        }
+
+        $this->output->writeln($string);
     }
 
     /**
