diff --git a/src/AppBundle/Tests/Controller/Api/ApiTestCase.php b/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
index bc82d69..4d35630 100644
--- a/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
+++ b/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
@@ -9,6 +9,7 @@ use GuzzleHttp\Message\AbstractMessage;
 use GuzzleHttp\Message\ResponseInterface;
 use GuzzleHttp\Subscriber\History;
 use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\Console\Output\ConsoleOutput;
 use Symfony\Component\DomCrawler\Crawler;
 
 class ApiTestCase extends KernelTestCase
@@ -25,6 +26,8 @@ class ApiTestCase extends KernelTestCase
      */
     protected $client;
 
+    private $output;
+
     public static function setUpBeforeClass()
     {
         self::$staticClient = new Client([
@@ -124,6 +127,10 @@ class ApiTestCase extends KernelTestCase
 
     protected function printDebug($string)
     {
-        echo $string."\n";
+        if ($this->output === null) {
+            $this->output = new ConsoleOutput();
+        }
+
+        $this->output->writeln($string);
     }
 }
\ No newline at end of file
