diff --git a/src/AppBundle/Tests/Controller/Api/ApiTestCase.php b/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
index 72ff59d..9b17bfb 100644
--- a/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
+++ b/src/AppBundle/Tests/Controller/Api/ApiTestCase.php
@@ -28,6 +28,11 @@ class ApiTestCase extends KernelTestCase
     protected $client;
 
     /**
+     * @var ConsoleOutput
+     */
+    private $output;
+
+    /**
      * @var FormatterHelper
      */
     private $formatterHelper;
@@ -179,7 +184,11 @@ class ApiTestCase extends KernelTestCase
      */
     protected function printDebug($string)
     {
-        echo $string."\n";
+        if ($this->output === null) {
+            $this->output = new ConsoleOutput();
+        }
+
+        $this->output->writeln($string);
     }
 
     /**
